code/datums/ai/basic_mobs/basic_ai_behaviors/targetting.dm

```diff
@@ -14,7 +14,7 @@
 		CRASH("No target datum was supplied in the blackboard for [controller.pawn]")
 
 	var/atom/current_target = controller.blackboard[target_key]
-	if (targetting_datum.can_attack(living_mob, current_target))
+	if (targetting_datum.can_attack(living_mob, current_target) && can_see(living_mob, current_target, vision_range))
```

Comment:
Why is this in `targetting` and not in `can_attack`?
You could then also more easily make it configurably optional per mob.